astronomer/providers/snowflake/hooks/snowflake.py

```diff
@@ -81,16 +83,28 @@ def run(  # type: ignore[override]
                 conn.commit()
         return self.query_ids
 
-    def check_query_output(self, query_ids: List[str]) -> None:
+    def check_query_output(
+        self, query_ids: List[str], handler: Callable | None = None, return_last: bool = True
+    ) -> Optional[Union[Any, List[Any]]]:
         """Once the query is finished fetch the result and log it in airflow"""
         with closing(self.get_conn()) as conn:
             with closing(conn.cursor(DictCursor)) as cur:
+                results = []
                 for query_id in query_ids:
                     cur.get_results_from_sfqid(query_id)
-                    cur.fetchall()
+                    if handler is not None:
+                        result = handler(cur)
+                        results.append(result)
                     self.log.info("Rows affected: %s", cur.rowcount)
                     self.log.info("Snowflake query id: %s", query_id)
 
+        if handler is None:
+            return None
+        elif return_last:
+            return results[-1]
```

Comment:
this looks weird to me, won't this would break if `handler` is None ?

Reply:
If the handler is None it won't fail it will return None from this function